samples/bluetooth/broadcast_assistant/src/main.c

```diff
@@ -0,0 +1,387 @@
+	.timeout = scan_timeout_cb,
+};
+
+static void scan_for_broadcast_source(void)
+{
+	int err;
+
+	scanning_for_broadcast_source = true;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning for Broadcast Source successfully started\n");
+
+	err = k_sem_take(&sem_source_discovered, K_FOREVER);
+	if (err != 0) {
+		printk("Failed to take sem_source_discovered (err %d)\n", err);
+	}
+}
+
+static void scan_for_broadcast_sink(void)
+{
+	int err;
+
+	scanning_for_broadcast_source = false;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, NULL);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning for Broadcast Sink successfully started\n");
+
+	err = k_sem_take(&sem_sink_discovered, K_FOREVER);
+	if (err != 0) {
+		printk("Failed to take sem_sink_discovered (err %d)\n", err);
+	}
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		printk("Failed to connect to %s (%u)\n", addr, err);
+
+		bt_conn_unref(broadcast_sink_conn);
+		broadcast_sink_conn = NULL;
+
+		scan_for_broadcast_sink();
+		return;
+	}
+
+	if (conn != broadcast_sink_conn) {
+		return;
+	}
+
+	printk("Connected: %s\n", addr);
+	k_sem_give(&sem_sink_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	if (conn != broadcast_sink_conn) {
+		return;
+	}
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	bt_conn_unref(broadcast_sink_conn);
+	broadcast_sink_conn = NULL;
+
+	k_sem_give(&sem_sink_disconnected);
+}
+
+static void security_changed_cb(struct bt_conn *conn, bt_security_t level,
+				enum bt_security_err err)
+{
+	if (err == 0) {
+		printk("Security level changed: %u\n", level);
+		k_sem_give(&sem_security_updated);
+	} else {
+		printk("Failed to set security level: %u\n", err);
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.security_changed = security_changed_cb
+};
+
+
+int main(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return 0;
+	}
+	printk("Bluetooth initialized\n");
+
+	bt_le_scan_cb_register(&scan_callbacks);
+
+	scan_for_broadcast_sink();
+
+	err = k_sem_take(&sem_sink_connected, K_FOREVER);
+	if (err != 0) {
+		printk("Failed to take sem_sink_connected (err %d)\n", err);
+	}
+
+	bt_bap_broadcast_assistant_discover(broadcast_sink_conn);
+
+	err = k_sem_take(&sem_security_updated, K_FOREVER);
+	if (err != 0) {
+		printk("Failed to take sem_security_updated (err %d)\n", err);
+	}
+
+	scan_for_broadcast_source();
+
+	struct bt_bap_scan_delegator_subgroup subgroup = { 0 };
+
+	subgroup.bis_sync = 0x06;
```

Comment:
Hardcoded value ;) 

To keep it simple, suggest to use `BT_BAP_BIS_SYNC_NO_PREF` to indicate that this device has no preference as to what the sink syncs to

Reply:
thanks - didn't know this one.  Will do a few tests before pushing though :)

Reply:
One thing I noticed here:  When the source has multiple streams (e.g. using the USB streaming Audio Source sample) together with the Samsung Buds, I need to set just one bit or the buds will never play.  I assume 0x06 = 110b => right + left ...

Reply:
As discussed offline, there is no relationsship between BIS indexes and locations :) If you want to provide any specific indexes, you need to parse the BASE for location information.